examples/access-control-migration/src/application.ts

```diff
@@ -0,0 +1,114 @@
+
+import {
+  AuthenticationComponent,
+  registerAuthenticationStrategy,
+} from '@loopback/authentication';
+import {
+  AuthorizationComponent,
+  AuthorizationTags,
+} from '@loopback/authorization';
+import {BootMixin} from '@loopback/boot';
+import {ApplicationConfig, BindingKey} from '@loopback/core';
+import {RepositoryMixin} from '@loopback/repository';
+import {RestApplication} from '@loopback/rest';
+import {RestExplorerComponent} from '@loopback/rest-explorer';
+import {ServiceMixin} from '@loopback/service-proxy';
+import path from 'path';
+import {
+  TokenServiceBindings,
+  TokenServiceConstants,
+  UserServiceBindings,
+} from './keys';
+import {MySequence} from './sequence';
+import {CasbinAuthorizationProvider} from './services/casbin.authorizer';
+import {getCasbinEnforcerByName} from './services/casbin.enforcers';
+import {JWTAuthenticationStrategy} from './services/jwt.auth.strategy';
+import {JWTService} from './services/jwt.service';
+import {SECURITY_SCHEME_SPEC} from './services/security.spec';
+import {MyUserService} from './services/user.service';
+
+/**
+ * Information from package.json
+ */
+export interface PackageInfo {
+  name: string;
+  version: string;
+  description: string;
+}
+export const PackageKey = BindingKey.create<PackageInfo>('application.package');
+
+export class AccessControlApplication extends BootMixin(
+  ServiceMixin(RepositoryMixin(RestApplication)),
+) {
+  constructor(options: ApplicationConfig = {}) {
+    super(options);
+
+    // Set up the custom sequence
+    this.sequence(MySequence);
+
+    // Set up default home page
+    this.static('/', path.join(__dirname, '../public'));
+
+    this.setUpBindings();
+    this.addSecuritySpec();
+
+    this.component(RestExplorerComponent);
+    // Bind authentication component related elements
+    this.component(AuthenticationComponent);
+    this.component(AuthorizationComponent);
+
+    this.projectRoot = __dirname;
+    // Customize @loopback/boot Booter Conventions here
+    this.bootOptions = {
+      controllers: {
+        // Customize ControllerBooter Conventions here
+        dirs: ['controllers'],
+        extensions: ['.controller.js'],
+        nested: true,
+      },
+    };
+  }
+
+  setUpBindings(): void {
+    this.bind(TokenServiceBindings.TOKEN_SECRET).to(
+      TokenServiceConstants.TOKEN_SECRET_VALUE,
+    );
+    this.bind(TokenServiceBindings.TOKEN_EXPIRES_IN).to(
+      TokenServiceConstants.TOKEN_EXPIRES_IN_VALUE,
+    );
+    this.bind(TokenServiceBindings.TOKEN_SERVICE).toClass(JWTService);
+
+    this.bind(UserServiceBindings.USER_SERVICE).toClass(MyUserService);
+
+    // authorization
+    this.bind('casbin.enforcer.factory').to(getCasbinEnforcerByName);
+    this.bind('authorizationProviders.casbin-provider')
+      .toProvider(CasbinAuthorizationProvider)
+      .tag(AuthorizationTags.AUTHORIZER);
+
+    // authentication
+    registerAuthenticationStrategy(this, JWTAuthenticationStrategy);
+  }
```

Comment:
Maybe we should move `setupBindings` to  a local component inside this app. This will show the possibility to have ACLs from an extension package in the future.

Reply:
@raymondfeng yeah moving the authentication setup to a component/extension is tracked in discussion https://github.com/strongloop/loopback-next/pull/4571#discussion_r383710046, I will create a follow-up story to extract the component.